devmand/gateway/src/devmand/channels/cli/SshSessionAsync.cpp

```diff
@@ -46,7 +51,7 @@ SshSessionAsync::~SshSessionAsync() {
 }
 
 Future<string> SshSessionAsync::read(int timeoutMillis) {
-  return via(executor.get(), [this, timeoutMillis] {
+  return via(serialExecutor.get(), [this, timeoutMillis] {
```

Comment:
Lambda captures this. Please use shard_from_this pattern to prevent segfaults when this is destructed before lambda is executed.

Reply:
ok done